### PR TITLE
Option for maximum y-axis

### PR DESCRIPTION
**Pull request by NikitaPoly**

### CODE REVIEW

File: docs/source/gettingstarted.rst [hunk 1]
```diff
@@ -179,6 +179,7 @@ Adding Arguments to write_xlsx()
 
 There are a few arguments you can add to the ``write_xlsx`` function to modify the ``.xlsx`` output file.
 They are:
+- ``max_y``
```
Comment (tannermpeterson) [inline docs/source/gettingstarted.rst:182]: Need to have a blank line before and after lists or sometimes sphinx won't render them correctly

File: docs/source/gettingstarted.rst [hunk 2]
```diff
@@ -188,6 +189,21 @@ They are:
 Any combination of these arguments can be given. Omitting them all is fine too.
 Their behavior is documented in detail below.
 
+``max_y``
+^^^^^^^^^
+
+Specifies the maximum active twitch force to be set in the output xlsx file.
```
Comment (tannermpeterson) [inline docs/source/gettingstarted.rst:195]: This line is bit confusing. It's clear from the following lines what this argument does so I think we can just remove this line

File: docs/source/gettingstarted.rst [hunk 3]
```diff
@@ -188,6 +189,21 @@ They are:
...
+When specified, max_y sets all graphs in continuous-waveform-snapshot and full-continuous-waveform-plots to have a y-axis scale of 0 - max_y.
```
Comment (tannermpeterson) [inline docs/source/gettingstarted.rst:196]: We should word this the same as the descriptions for the other args:

`Specifies the max y-axis bound for all graphs in continuous-waveform-snapshot and full-continuous-waveform-plots. The min bound is always 0. By default, this parameter is set to the max tissue waveform value across all wells.`

**Review by tannermpeterson — CHANGES_REQUESTED** [review]

File: src/pulse3D/excel_writer.py [hunk 4]
```diff
@@ -1,4 +1,5 @@
 # -*- coding: utf-8 -*-
+from asyncio import constants
```
Comment (tannermpeterson) [inline src/pulse3D/excel_writer.py:2]: Unused import

File: src/pulse3D/excel_writer.py [hunk 5]
```diff
@@ -582,7 +594,7 @@ def create_waveform_charts(
     full_plot_params = plotting_parameters(dm["end_time"] - dm["start_time"])
 
     full_chart.set_x_axis({"name": "Time (seconds)", "min": dm["start_time"], "max": dm["end_time"]})
-    full_chart.set_y_axis({"name": "Active Twitch Force (μN)", "major_gridlines": {"visible": 0}})
+    full_chart.set_y_axis({"name": "Active Twitch Force (μN)", "major_gridlines": {"visible": 0},"max":max_y})
```
Comment (tannermpeterson) [inline src/pulse3D/excel_writer.py:597]: Noticing this and a few other minor formatting issues with whitespace

File: src/pulse3D/excel_writer.py [hunk 6]
```diff
@@ -313,6 +316,7 @@ def write_xlsx(
 
     log.info("Computing data metrics for each well.")
 
+    max_activeTwitchForce = 0
```
Comment (tannermpeterson) [inline src/pulse3D/excel_writer.py:319]: Use `snake_case` not `camelCase`. You can also just make this `max_force_of_recording` to make it more clear what's included in this max

File: src/pulse3D/excel_writer.py [hunk 7]
```diff
@@ -359,7 +363,9 @@ def write_xlsx(
         min_value = min(interpolated_force)
         interpolated_force -= min_value
         interpolated_force *= MICRO_TO_BASE_CONVERSION
-
+        #find the biggest activation twitch force over all
+        if max(interpolated_force) > max_activeTwitchForce:
```
Comment (tannermpeterson) [inline src/pulse3D/excel_writer.py:367]: A simple pattern for tracking a max value is `max_val = max(max_val, new_val)`. You can make this simpler and more clear by removing the if branch and just doing
```
max_force_of_well = max(interpolated_force)
max_force_of_recording = max(max_force_of_recording, max_force_of_well)
```

**Review by tannermpeterson — CHANGES_REQUESTED** [review]

File: CHANGELOG.rst [hunk 8]
```diff
@@ -1,5 +1,8 @@
 Changelog for Pulse3D
 =====================
+0.25.0
```
Comment (tannermpeterson) [inline CHANGELOG.rst:3]: Need to include release date here. Also just note that the patch version can go past 9 (i.e. 0.24.10 is ok), but I think using 0.25.0 here is fine

**Review by tannermpeterson — CHANGES_REQUESTED** [review]

File: CHANGELOG.rst [hunk 9]
```diff
@@ -1,5 +1,8 @@
 Changelog for Pulse3D
 =====================
+0.25.0 (2022-23-22)
```
Comment (tannermpeterson) [inline CHANGELOG.rst:3]: 2022-08-23

**Review by tannermpeterson — CHANGES_REQUESTED** [review]

File: CHANGELOG.rst (hunk 9 above)
Comment (NikitaPoly) [reply CHANGELOG.rst:3]: I keep forgetting that the yyyy-mm-dd is standard not yyyy-dd-mm

File: CHANGELOG.rst [hunk 10]
```diff
@@ -1,5 +1,8 @@
 Changelog for Pulse3D
 =====================
+0.25.0 (2022-08-23)
+- Added the option to set custom y-axis for output graphs
+-------------------
```
Comment (tannermpeterson) [inline CHANGELOG.rst:5]: Sorry last thing, this `-------------------` needs to be moved up one line

**Review by tannermpeterson — CHANGES_REQUESTED** [review]

File: CHANGELOG.rst (hunk 10 above)
Comment (NikitaPoly) [reply CHANGELOG.rst:5]: I will pay attention to the formatting of variables and logs more.

**Review by tannermpeterson — APPROVED** [review]

**Pull request merged**